File: legacy/pasarguardbot/frontend/src/types/panel/payments.ts
/** Mirrors app/models/panel/payments.py */
import type { PanelAuthRequest, PanelEnvelope } from "./common";

export const WALLET_TYPES = ["TRX", "USDT", "TON"] as const;

export interface PanelWalletRow {
  id: number;
  type: string;
  address: string;
  has_api_key: boolean;
}

export interface PanelCardRow {
  id: number;
  number: string;
  name: string;
  active: boolean;
}

export interface PanelAutoApproveRuleRow {
  id: number;
  min_successful_tx: number;
  max_successful_tx?: number | null;
  auto_approve_delay_minutes: number;
  is_active: boolean;
}

export interface PanelPaymentsResponse extends PanelEnvelope {
  wallets: PanelWalletRow[];
  available_wallet_types: string[];
  cards: PanelCardRow[];
  rules: PanelAutoApproveRuleRow[];
}

export interface PanelWalletCreateRequest extends PanelAuthRequest {
  wallet_type: string;
  address: string;
  api_key?: string;
}

export interface PanelWalletDeleteRequest extends PanelAuthRequest {
  wallet_id: number;
}

export interface PanelCardCreateRequest extends PanelAuthRequest {
  number: string;
  name: string;
  active?: boolean;
}

export interface PanelCardActionRequest extends PanelAuthRequest {
  card_id: number;
}

export interface PanelRuleCreateRequest extends PanelAuthRequest {
  min_successful_tx?: number;
  max_successful_tx?: number | null;
  auto_approve_delay_minutes?: number;
}

export interface PanelRuleToggleRequest extends PanelAuthRequest {
  rule_id: number;
  is_active: boolean;
}

export interface PanelRuleDeleteRequest extends PanelAuthRequest {
  rule_id: number;
}

File: legacy/pasarguardbot/app/models/panel/payments.py
"""Admin panel DTOs: crypto wallets, manual cards and auto-approve rules."""

from pydantic import BaseModel, Field

from app.models.panel.common import PanelRequest, PanelResponse

WALLET_TYPES = ("TRX", "USDT", "USDT-TON", "USDT-BEP20", "TON", "POL")


class PanelWalletRow(BaseModel):
    id: int
    type: str
    address: str
    has_api_key: bool = False


class PanelCardRow(BaseModel):
    id: int
    number: str
    name: str
    active: bool = False


class PanelAutoApproveRuleRow(BaseModel):
    id: int
    min_successful_tx: int = 0
    max_successful_tx: int | None = None
    auto_approve_delay_minutes: int = 0
    is_active: bool = False


class PanelPaymentsResponse(PanelResponse):
    wallets: list[PanelWalletRow] = Field(default_factory=list)
    available_wallet_types: list[str] = Field(default_factory=list)
    cards: list[PanelCardRow] = Field(default_factory=list)
    rules: list[PanelAutoApproveRuleRow] = Field(default_factory=list)


class PanelWalletCreateRequest(PanelRequest):
    wallet_type: str
    address: str = Field(..., min_length=4, max_length=128)
    api_key: str = Field("", max_length=128)


class PanelWalletDeleteRequest(PanelRequest):
    wallet_id: int


class PanelCardCreateRequest(PanelRequest):
    number: str = Field(..., min_length=12, max_length=32)
    name: str = Field(..., min_length=1, max_length=64)
    active: bool = False


class PanelCardActionRequest(PanelRequest):
    card_id: int


class PanelRuleCreateRequest(PanelRequest):
    min_successful_tx: int = Field(0, ge=0)
    max_successful_tx: int | None = Field(None, ge=0)
    auto_approve_delay_minutes: int = Field(30, ge=0)


class PanelRuleToggleRequest(PanelRequest):
    rule_id: int
    is_active: bool


class PanelRuleDeleteRequest(PanelRequest):
    rule_id: int
